### PR TITLE
Fix Script Commands NPC Last

### PR DESCRIPTION
**Pull request by sammeepay**
Fix Script Commands NPC Last

### CODE REVIEW

File: npc/re/merchants/enchan_mal.txt [hunk 1]
```diff
@@ -22,662 +22,636 @@
...
 		specialeffect2 EF_REPAIRWEAPON;
 		mes "[Mayomayo]";
-		mes "Initialize the enchant effect from the equipment.";
-		delitem 6417,1;	// Silvervine
+		mes "ź�Ϳ�͡�ҡ���������.";
+		delitem 6417,1; //Silvervine
 
 		// anti-hack
-		if (callfunc("F_IsEquipIDHack", EQI_HAND_R, .@equip_id) || callfunc("F_IsEquipRefineHack", EQI_HAND_R, .@equip_refine) || 
+		if (callfunc("F_IsEquipIDHack", EQI_HAND_R, .@equip_id) ||
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mal.txt:644]: Fix
```suggestion
		if (callfunc("F_IsEquipIDHack", EQI_HAND_R, .@equip_id) || callfunc("F_IsEquipRefineHack", EQI_HAND_R, .@equip_refine) || 
```

File: npc/re/merchants/enchan_mal.txt [hunk 2]
```diff
@@ -22,662 +22,636 @@
...
 		delequip EQI_HAND_R;
 
-		// GetNonSlotItemSock2 .@equip_refine .@equip_id .@equip_card[0] .@equip_card[1] .@equip_card[2] .@equip_card[3]
-		for ( .@i = getiteminfo(.@equip_id,10); .@i < MAX_SLOTS; .@i++ ) {
-			if (callfunc("F_IsCharm",.@equip_card[.@i]) == true)
-				.@equip_card[.@i] = 0;// Armor Enchant System
+//		GetNonSlotItemSock2 .@equip_refine .@equip_id .@equip_card[0] .@equip_card[1] .@equip_card[2] .@equip_card[3]
+		for(set .@i,0; .@i<4; set .@i,.@i+1) {
+			if (.@equip_card[.@i] >= 4700) // Armor Enchant System
+				set .@equip_card[.@i],0;
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mal.txt:654]: Fix
```suggestion
		// GetNonSlotItemSock2 .@equip_refine .@equip_id .@equip_card[0] .@equip_card[1] .@equip_card[2] .@equip_card[3]
		for ( .@i = getiteminfo(.@equip_id,10); .@i < MAX_SLOTS; .@i++ ) {
			if (callfunc("F_IsCharm",.@equip_card[.@i]) == true)
				.@equip_card[.@i] = 0;// Armor Enchant System
```

File: npc/re/merchants/enchan_mora.txt [hunk 3]
```diff
@@ -21,66 +21,66 @@
...
 			close;
 		case 2:
 			if (countitem(2864) || countitem(2865) || countitem(2866)) {
 				mes "[Artifice]";
-				mes "You are already allowed access to our relics by your possession of a Ring!";
+				mes "��Ҩ��š����¹����������ǹ�������������!";
 				close;
 			}
-			if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Arch_Bishop) {
+			if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Bishop) {
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:78]: Fix
```suggestion
			if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Arch_Bishop) {
```

File: npc/re/merchants/enchan_mora.txt [hunk 4]
```diff
@@ -103,82 +103,82 @@ mora,88,89,5	script	Keeper of Secrets#pa082	495,{
 		delitem 6380,10; //Mora_Coin
 		getitem .@item,1;
 		mes "[Artifice]";
-		mes "You have obtained ^aa00aa"+getitemname(.@item)+"^000000. May the Blessing of the Light be with you.";
+		mes "������Ѻ ^aa00aa"+getitemname(.@item)+"^000000. ������ʧ��¾������.";
 		close;
 	case 3:
-		if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Arch_Bishop) {
+		if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Bishop) {
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:109]: Fix
```suggestion
		if (Class == Job_Arch_Bishop || Class == Job_Arch_Bishop_T || Class == Job_Baby_Arch_Bishop) {
```

File: npc/re/merchants/enchan_mora.txt [hunk 5]
```diff
@@ -1139,19 +1139,14 @@ L_Socket:
...
-			mes "You've got to pay if you want the services.";
+			mes "��ҵ�ͧ���¤�Һ�ԡ�ô���.";
 			close;
 		}
 		specialeffect2 EF_REPAIRWEAPON;
 		mes "[Artifact Researcher]";
-		mes "The ability to enhance remains.";
+		mes "ź�Ϳ���º����.";
 		delitem 6380,1; //Mora_Coin
 		set Zeny, Zeny-100000;
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1149]: Fix
```suggestion
		set Zeny, Zeny-100000;

		// anti-hack
		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) || callfunc("F_IsEquipCardHack", .@part, .@equip_card[0], .@equip_card[1], .@equip_card[2], .@equip_card[3]) || callfunc("F_IsEquipRefineHack", .@part, .@equip_refine))
			close;

```

File: npc/re/merchants/enchan_mora.txt [hunk 6]
```diff
@@ -1312,32 +1307,31 @@ mora,124,82,4	script	Artifact Collector#blan	522,{
...
 		emotion ET_FRET;
 		mes "[Artifact Collector]";
-		mes "I don't know what you are trying to do... I can't enhance what you don't have.";
+		mes "��������������Ҿ�����з�����... ������������ѹ�Ҵ���.";
 		close;
 	}
 	set .@equip_id, getequipid(EQI_HAND_R);
-	.@equip_refine = getequiprefinerycnt(EQI_HAND_R);
-	setarray .@equip_card[0], getequipcardid(EQI_HAND_R,0), getequipcardid(EQI_HAND_R,1), getequipcardid(EQI_HAND_R,2),getequipcardid(EQI_HAND_R,3);
+	setarray .@equip_card[2], getequipcardid(EQI_HAND_R,2),getequipcardid(EQI_HAND_R,3);
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1334]: Fix
```suggestion
	.@equip_refine = getequiprefinerycnt(EQI_HAND_R);
	setarray .@equip_card[0], getequipcardid(EQI_HAND_R,0), getequipcardid(EQI_HAND_R,1), getequipcardid(EQI_HAND_R,2),getequipcardid(EQI_HAND_R,3);
```

File: npc/re/merchants/enchan_mora.txt [hunk 7]
```diff
@@ -1362,130 +1356,125 @@ mora,124,82,4	script	Artifact Collector#blan	522,{
 	default:
 		emotion ET_FRET;
 		mes "[Artifact Collector]";
-		mes "The weapon you have isn't one of the Artifacts I can work with.";
+		mes "���ظ�������������������觻�д�ɰ����ҵ�ͧ���.";
 		close;
 	}
-	if (.@equip_refine < 7) {
+	if (getequiprefinerycnt(EQI_HAND_R) < 7) {
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1362]: Fix
```suggestion
	if (.@equip_refine < 7) {
```

File: npc/re/merchants/enchan_mora.txt [hunk 8]
```diff
@@ -1362,130 +1356,125 @@ mora,124,82,4	script	Artifact Collector#blan	522,{
...
+		if(select("���, ����������.:��, �����!") == 1) {
 			emotion ET_FRET;
 			mes "[Artifact Collector]";
-			mes "You've got to be kidding... you got my hopes up...";
+			mes "���������... �����������ѧ���...";
 			close;
 		}
 		break;
 	}
 	emotion ET_SMILE;
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1407]: Fix
```suggestion
	emotion ET_SMILE;

	// anti-hack
	if (callfunc("F_IsEquipIDHack", EQI_HAND_R, .@equip_id) || callfunc("F_IsEquipCardHack", EQI_HAND_R, .@equip_card[0], .@equip_card[1], .@equip_card[2], .@equip_card[3]) || callfunc("F_IsEquipRefineHack", EQI_HAND_R, .@equip_refine))
		close;

```

File: npc/re/merchants/enchan_mora.txt [hunk 9]
```diff
@@ -1498,8 +1487,6 @@ function	script	F_Mora_Enchant	{
 	specialeffect2 EF_REPAIRWEAPON;
 	progressbar "ffff00",3;
 	delitem 6380,5; //Mora_Coin
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1489]: Fix
```suggestion
	delitem 6380,5; //Mora_Coin
	if (countitem(getarg(0)) < 1)
		return 0;
```

File: npc/re/merchants/enchan_mora.txt [hunk 10]
```diff
@@ -1529,252 +1516,249 @@ function	script	F_Mora_Enchant	{
 	return 1;
 }
 mora,105,176,4	script	Master Tailor#pa0829	509,{
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1518]: Fix
```suggestion
mora,105,176,4	script	Master Tailor#pa0829	509,{
	disable_items;
```

File: npc/re/merchants/enchan_mora.txt [hunk 11]
```diff
@@ -1529,252 +1516,249 @@ function	script	F_Mora_Enchant	{
...
 			}
 			mes "[Master Tailor]";
-			mes "My art looks great.";
+			mes "�ŧҹ����ʹ����������.";
 			close;
 		}
 	}
 }
 
 mora,123,177,6	script	Pendant Crafter#pa0829	509,{
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1610]: Fix
```suggestion
mora,123,177,6	script	Pendant Crafter#pa0829	509,{
	disable_items;
```

File: npc/re/merchants/enchan_mora.txt [hunk 12]
```diff
@@ -1529,252 +1516,249 @@ function	script	F_Mora_Enchant	{
...
 			}
 			mes "[Pendant Crafter]";
-			mes "Looks like I did it!";
+			mes "������͹��Ң�Ҩз���!";
 			close;
 		}
 	}
 }
 
 mora,134,166,4	script	Bulberry Westhood#pa0829	509,{
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_mora.txt:1680]: Fix
```suggestion
mora,134,166,4	script	Bulberry Westhood#pa0829	509,{
	disable_items;
```

File: npc/re/merchants/enchan_rockridge.txt [hunk 13]
```diff
@@ -340,89 +340,89 @@ har_in01,34,81,5	script	Howard#rr	4_M_TATIO,{
...
 	getitem atoi(.@data$[.@s+1]),1;
 	close;
 }
 
--	shop	shotguns_rockridge	-1,13154:-1,13155:-1,13192:-1,13193:-1,13194:-1
--	shop	revolvers_rockridge	-1,13102:-1,13120:-1,13122:-1
--	shop	bullets_rockridge	-1,13200:-1,13221:-1,13222:-1,13228:-1,13229:-1,13230:-1,13231:-1,13232:-1
+-	shop	shotguns_rockridge	-1,13154:12000,13155:32000,13192:1250000,13193:1350000,13194:1950000
+-	shop	revolvers_rockridge	-1,13102:20000,13120:1200000,13122:1450000
+-	shop	bullets_rockridge	-1,13200:1,13221:-1,13222:10,13228:10,13229:10,13230:10,13231:10,13232:10
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_rockridge.txt:367]: Fix
```suggestion
-	shop	shotguns_rockridge	-1,13154:-1,13155:-1,13192:-1,13193:-1,13194:-1
-	shop	revolvers_rockridge	-1,13102:-1,13120:-1,13122:-1
-	shop	bullets_rockridge	-1,13200:-1,13221:-1,13222:-1,13228:-1,13229:-1,13230:-1,13231:-1,13232:-1
```

File: npc/re/merchants/enchan_rockridge.txt [hunk 14]
```diff
@@ -340,89 +340,89 @@ har_in01,34,81,5	script	Howard#rr	4_M_TATIO,{
...
-	mes "But ^0000CDall my products are hard to find elsewhere^000000. Why don't you take a look at them?";
+	mes "��д����˵ؼ����ǡѹ, ����������ö ^0000CDŴ�Ҥ������^000000.";
+	mes "�� ^0000CD�Թ��ҷ�����ͧ����������ö���������^000000. �����������ͧ���ѡ˹��� ?";
 	close2;
 	callshop "armor_rockridge", 1;
 	end;
 }
 
-har_in01,34,84,3	shop	Tool Dealer#rockridge00	1_F_03,611:-1,645:-1,656:-1,601:-1,602:-1,1065:-1,610:-1
+har_in01,34,84,3	shop	Tool Dealer#rockridge00	1_F_03,611:40,645:800,656:10500,601:60,602:300,1065:100,610:4000
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_rockridge.txt:428]: Fix
```suggestion
har_in01,34,84,3	shop	Tool Dealer#rockridge00	1_F_03,611:-1,645:-1,656:-1,601:-1,602:-1,1065:-1,610:-1
```

File: npc/re/merchants/enchan_upg.txt [hunk 15]
```diff
@@ -75,33 +75,32 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 	}
 	setarray .@equip_card[0], getequipcardid(.@part,0), getequipcardid(.@part,1), getequipcardid(.@part,2), getequipcardid(.@part,3);
-	// if (!getequipisequiped(.@part)) {
-	// 	mes "It is dangerous to remove equipment during enchant process!";
-	// 	close;
-	// }
+	if (!getequipisequiped(.@part)) {
+		mes "�ѹ�ѹ���¶�����������ظ�͡��С��ѧ�Ϳ!";
+		close;
+	}
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:85]: Fix
```suggestion
	// if (!getequipisequiped(.@part)) {
	// 	mes "It is dangerous to remove equipment during enchant process!";
	// 	close;
	// }
```

File: npc/re/merchants/enchan_upg.txt [hunk 16]
```diff
@@ -75,33 +75,32 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 	set .@equip_id, getequipid(.@part);
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:86]: Fix
```suggestion
	set .@equip_id, getequipid(.@part);
	.@equip_refine = getequiprefinerycnt(.@part);
```

File: npc/re/merchants/enchan_upg.txt [hunk 17]
```diff
@@ -206,31 +205,32 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 		} else {
 			specialeffect EF_REPAIRWEAPON;
-			mes "The slot ^9900004^000000 has been enchanted!";
+			mes "��ͧ��� ^9900004^000000 ���Ϳ����!";
 		}
 		delitem 6484,1; //Enchant_Book
 
 		// anti-hack
-		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) || callfunc("F_IsEquipRefineHack", .@part, .@equip_refine) || 
+		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) ||
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:233]: Fix
```suggestion
		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) || callfunc("F_IsEquipRefineHack", .@part, .@equip_refine) || 
```

File: npc/re/merchants/enchan_upg.txt [hunk 18]
```diff
@@ -242,39 +242,40 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 		close;
 	} else if (.@select == 2) {
 		if (Zeny < 100000) {
-			mes "You need to bring some money to initialize!!";
+			mes "��ҵ�ͧ���Թ����ź�Ϳ!!";
 			close;
 		}
-		if (callfunc("F_IsCharm",.@equip_card[3]) == false) {
-			mes "This item is not enchanted!";
+		if (.@equip_card[3] < 4700) { // Armor Enchant System
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:253]: Fix
```suggestion
		if (callfunc("F_IsCharm",.@equip_card[3]) == false) {
```

File: npc/re/merchants/enchan_upg.txt [hunk 19]
```diff
@@ -242,39 +242,40 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 			close;
 		}
 		specialeffect EF_REPAIRWEAPON;
-		mes "I initialized the enchant effects.";
+		mes "���ź�Ϳ�������.";
 		set Zeny, Zeny - 100000;
 
 		// anti-hack
-		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) || callfunc("F_IsEquipRefineHack", .@part, .@equip_refine) || 
+		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) ||
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:266]: Fix
```suggestion
		if (callfunc("F_IsEquipIDHack", .@part, .@equip_id) || callfunc("F_IsEquipRefineHack", .@part, .@equip_refine) || 
```

File: npc/re/merchants/enchan_upg.txt [hunk 20]
```diff
@@ -242,39 +242,40 @@ prt_in,28,73,3	script	Devil Enchant Master#prq	63,{
...
 		    callfunc("F_IsEquipCardHack", .@part, .@equip_card[0], .@equip_card[1], .@equip_card[2], .@equip_card[3]))
 			close;
 
+		set .@equip_refine, getequiprefinerycnt(.@part);
```
Comment (Azael-Dev) [inline npc/re/merchants/enchan_upg.txt:270]: Remove
```suggestion

```

File: npc/re/merchants/moro_cav_exchange.txt [hunk 21]
```diff
@@ -1,98 +1,98 @@
 //===== rAthena Script =======================================
 //= Enchants NPCs.
 //===== Description: =========================================
-//= [Official Conversion]
-//= Hero Ring enchants NPCs
+//= Hero Ring enchants NPCs - Walkthrough Conversion.
```
Comment (Azael-Dev) [inline npc/re/merchants/moro_cav_exchange.txt:4]: Fix
```suggestion
//= [Official Conversion]
//= Hero Ring enchants NPCs
```

File: npc/re/merchants/moro_cav_exchange.txt [hunk 22]
```diff
@@ -1,98 +1,98 @@
...
 //===== Changelogs: ==========================================
 //= 1.0 First Version. [Capuche]
```
Comment (Azael-Dev) [inline npc/re/merchants/moro_cav_exchange.txt:6]: Fix
```suggestion
//= 1.0 First Version. [Capuche]
//= 1.1 Slight change to match official script. [Capuche]
```

File: npc/re/merchants/moro_cav_exchange.txt [hunk 23]
```diff
@@ -1,98 +1,98 @@
...
 moro_cav,32,73,5	script	Hero Token Exchanger#1	4_F_SHADOWCHASER,{
 	disable_items;
+	if (checkweight(1201,1) == 0) {// custom
+		mes "����բͧ�ҡ�Թ�. �ͧŴ�ӹǹ�ͧŧ.";
+		close;
+	}
+	if ((MaxWeight - Weight) < 10000) {
+		mes "����չ��˹ѡ�ҡ�Թ�, ������������ö���������. �ͧŴ���˹ѡŧ.";
+		close;
+	}
```
Comment (Azael-Dev) [inline npc/re/merchants/moro_cav_exchange.txt:18]: Remove
```suggestion

```

File: npc/re/merchants/moro_cav_exchange.txt [hunk 24]
```diff
@@ -1,98 +1,98 @@
...
 	while(true) {
 		next;
-		if (checkweight(2163,2) == 0) {
-			mes "[Hero Token Exchanger]";
-			mes "Your pack has a lot of items,";
-			mes "Please come again after arrange it!";
-			close;
-		}
-		switch( select( "Ring description", "Exchange with the ring.", "End conversation." ) ) {
+		switch( select( "��������´��ǹ", "�š��ǹ.", "����ʹ���." ) ) {
```
Comment (Azael-Dev) [inline npc/re/merchants/moro_cav_exchange.txt:27]: Fix
```suggestion
		if (checkweight(2163,2) == 0) {
			mes "[Hero Token Exchanger]";
			mes "Your pack has a lot of items,";
			mes "Please come again after arrange it!";
			close;
		}
		switch( select( "��������´��ǹ", "�š��ǹ.", "����ʹ���." ) ) {
```

**Comment by sammeepay** [comment]
Fix completed

**Pull request merged**